src/substrait/textplan/converter/HierarchyStack.cpp

```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/textplan/converter/HierarchyStack.h"
+
+namespace io::substrait::textplan {
+
+const ::google::protobuf::Message* HierarchyStack::getParent() {
+  if (stack_.empty())
+    return nullptr;
+  return stack_[stack_.size()];
```

Comment:
I don't think this is right is it?  wouldn't `stack_[stack_.size()]` be invalid?

Reply:
Ping

Reply:
Done.